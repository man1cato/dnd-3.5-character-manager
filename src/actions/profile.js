import database from '../firebase/firebase'


//SET PROFILE IN STORE
export const setProfile = (id, profile) => ({
    type: 'SET_PROFILE',
    id,
    profile
});

//READ PROFILE FROM FIREBASE
export const startSetProfile = (id) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid;
        return database.ref(`users/${uid}/profiles/${id}`).once('value').then((snapshot) => {
            const profile = snapshot.val();
            dispatch(setProfile(id, profile));
        });
    };
};

//CREATE PROFILE IN FIREBASE
export const startCreateProfile = (profile) => {    
    return (dispatch, getState) => {
        const uid = getState().auth.uid       
        return database.ref(`users/${uid}/profiles`).push(profile).then((ref) => {
            dispatch(setProfile(ref.key, profile))
        })
    }
}


//UPDATE PROFILE IN STORE
export const editProfile = (updates) => ({
    type: 'EDIT_PROFILE',
    updates
});

//UPDATE PROFILE IN FIREBASE
export const startEditProfile = (id, updates) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid;
        return database.ref(`users/${uid}/profiles/${id}`).update(updates).then(() => {
            dispatch(editProfile(updates));
        });
    };
};

//DELETE PROFILE FROM STORE
export const deleteProfile = () => ({
    type: 'DELETE_PROFILE'
})

//LEVEL UP CHARACTER IN STORE
// export const levelUp = (character) => ({
//     type: 'LEVEL_UP',
//     character
// });
