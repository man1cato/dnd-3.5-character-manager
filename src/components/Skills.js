import React from 'react';

const Skills = (props) => (
  <div>
    <div className="grid grid--skills">
      <h4>Skill</h4>
      <h4 className="grid--skills__ranks">Ranks</h4>
      <h4>Skill</h4>
      <h4 className="grid--skills__ranks">Ranks</h4>
    </div>
    <div className="flexbox flexbox--skills">
      {props.skills.map((skill, i) => (
        <div className="flexbox flexbox--skills-item" key={i}>
          <div className="flexbox--skills-item__name">{skill.name}</div>
          <div className="flexbox--skills-item__ranks">{skill.ranks}</div>
        </div>
      ))}
    </div>
  </div>
)

export default Skills;


// <div className="grid grid--skills">
//   <h4 className="grid--skills__name">Skill</h4>
//   <h4 className="grid--skills__ranks">Ranks</h4>
//
//   {props.skills.map((skill, i) => (
//     <div className="grid--skills__name" key={i}>{skill.name}</div>
//   ))}
//   {props.skills.map((skill, i) => (
//     <div className="grid--skills__ranks" key={i}>{skill.ranks}</div>
//   ))}
// </div>
