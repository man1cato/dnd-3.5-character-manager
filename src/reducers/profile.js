export default (state = {}, action) => {
    switch (action.type) {
        case 'SET_PROFILE':
            return {
                id: action.id,
                ...action.fields
            };
        case 'EDIT_PROFILE':
            return {
                ...state,
                ...action.updates
            };
        default:
            return state;
    }
};
