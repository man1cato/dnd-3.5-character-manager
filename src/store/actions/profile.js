import database from '../../firebase/firebase'
import { startGetProfiles } from './profiles'

//SET SELECTED PROFILE IN STORE
export const setProfile = (id, profile) => ({
    type: 'SET_PROFILE',
    id,
    profile
})

//READ PROFILE FROM FIREBASE
export const startSetProfile = (uid, id) => {
    return (dispatch) => {
        return database.ref(`users/${uid}/profiles/${id}`).once('value').then((snapshot) => {
            const profile = snapshot.val()
            dispatch(setProfile(id, profile))
        })
    }
}

//CREATE PROFILE IN FIREBASE
export const startCreateProfile = (profile) => {    
    return (dispatch, getState) => {
        const uid = getState().auth.uid       
        return database.ref(`users/${uid}/profiles`).push(profile).then((ref) => {
            dispatch(setProfile(ref.key, profile))
            dispatch(startGetProfiles(uid))
            localStorage.setItem('selectedCharacterId', ref.key)
        })
    }
}


//UPDATE PROFILE IN STORE
export const editProfile = (updates) => ({
    type: 'EDIT_PROFILE',
    updates
})

//UPDATE PROFILE IN FIREBASE
export const startEditProfile = (id, updates) => {
    return (dispatch, getState) => {
        const uid = getState().auth.uid
        return database.ref(`users/${uid}/profiles/${id}`).update(updates).then(() => {
            dispatch(editProfile(updates))
        })
    }
}

//REMOVE PROFILE FROM STORE
export const removeProfile = () => ({
    type: 'REMOVE_PROFILE'
})

//REMOVE PROFILE FROM FIREBASE
// export const startRemoveProfile = () => {
//     return (dispatch) => {
//         return database.ref(`users/${uid}/profiles/${id}`).remove().then(() => {
//             dispatch(removeProfile())
//         })
//     }
// }

//LEVEL UP CHARACTER IN STORE
// export const levelUp = (character) => ({
//     type: 'LEVEL_UP',
//     character
// });
