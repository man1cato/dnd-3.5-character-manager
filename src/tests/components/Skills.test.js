import React from 'react';
import {shallow} from 'enzyme';

import Skills from '../../components/Skills';
import profile from '../fixtures/profile';

const profileData = {
  id: profile.id,
  ...profile.fields
};

test('should render skills with profile data', () => {
  const wrapper = shallow(<Skills {...profileData} />);
  expect(wrapper).toMatchSnapshot();
})
