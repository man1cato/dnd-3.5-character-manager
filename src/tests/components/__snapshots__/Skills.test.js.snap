// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render skills with profile data 1`] = `
<div>
  <div
    className="grid grid--skills"
  >
    <h4>
      Skill
    </h4>
    <h4
      className="grid--skills__ranks"
    >
      Ranks
    </h4>
    <h4>
      Skill
    </h4>
    <h4
      className="grid--skills__ranks"
    >
      Ranks
    </h4>
  </div>
  <div
    className="flexbox flexbox--skills"
  >
    <div
      className="flexbox flexbox--skills-item"
      key="0"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="1"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="2"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="3"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="4"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="5"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="6"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="7"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="8"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="9"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="10"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="11"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="12"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="13"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="14"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="15"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="16"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="17"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="18"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="19"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="20"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="21"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
    <div
      className="flexbox flexbox--skills-item"
      key="22"
    >
      <div
        className="flexbox--skills-item__name"
      />
      <div
        className="flexbox--skills-item__ranks"
      />
    </div>
  </div>
</div>
`;
